Drivers/RcReceiver/Src/sbus_receiver.cpp

```diff
@@ -0,0 +1,100 @@
+#include "sbus_receiver.hpp"
+#include "drivers_config.hpp"
+
+SBUSReceiver::SBUSReceiver(UART_HandleTypeDef* uart) : uart_(uart)
+{
+    for(int i = 0; i < SBUS_INPUT_CHANNELS; i++)
+    {
+        received_sbus_.ch[i] = 1000;
+    }
+    received_sbus_.ch17 = false;
+    received_sbus_.ch18 = false;
+    received_sbus_.failsafe = false;
+    received_sbus_.lost_frame = false;
+    received_sbus_.new_data = false;
+
+    HAL_UART_Receive_DMA (uart_, raw_sbus_, SBUS_FRAME_SIZE);
+}
+
+SBus_t SBUSReceiver::GetSBUS(){
+	if(received_sbus_.new_data == false)
+		HAL_UART_Receive_DMA (uart_, raw_sbus_, SBUS_FRAME_SIZE);
+    return received_sbus_;
+}
+
+RCControl SBUSReceiver::GetRCControl(){
+	if(received_sbus_.new_data == false)
+		HAL_UART_Receive_DMA (uart_, raw_sbus_, SBUS_FRAME_SIZE);
+    cast_rccontrol();
+    return received_rccontrol_;
+}
+
+void SBUSReceiver::parse()
```

Comment:
What exactly does this function do?

Reply:
Looks like it parses the raw byte data to create the rx channels